src/ReinforcementLearning/SimpleAgent/Agent.h

```diff
@@ -0,0 +1,89 @@
+#ifndef AGENT_H
+#define AGENT_H
+
+#include "RandomPolicy.h"
+#include "Policy.h"
+#include "../../alignment/State.h"
+#include <stdlib.h>
+#include <random>
+#include <chrono>
+#include <utility>
+#include <iostream>
+#include <map>
+#include "../../alignment/Graph.h"
+#include "Episode.h"
+
+using std::vector;
+
+/**TODO: Include score (maybe expected score) in history.
+ *       The policy is supposed to map states to a vector of action probabilities in my view.
+ */
+
+/** \brief This Agent class selects edges according to a policy.
+*/
+class Agent {
+    public:
+    state* constState;
+    Policy* policy;
+    Graph graph;
+    Agent(){};
+    Agent(state* cs, Graph& g, Policy* pol){
+        constState = cs;
+        policy = pol;
+        graph = g;
+    }
+    ~Agent(){};
+    /** This vector consists of pairs of selectedSubset (i.e. a state) and the action taken in that state.
+     */
+    std::vector <std::pair<std::vector <bool>, unsigned int>> history;
+    /** This function runs an episode. It relies on constState, policy and graph
+     *  to calculate state-action pairs as well as score.
+     * \return This function returns an episode
+     */
+    Episode getEpisode() {
+
+        Episode episode;
+        unsigned int n = constState->edges.size();
+        vector <bool> action(n, false);
+        episode.states.push_back(constState->selectedSubset); 
+        unsigned int counter = 0;
+        std::pair <state*, unsigned int> stateAction = executePolicy(constState, policy);
+        
```

Comment:
I think this does not work when sampling multiple Episodes (what we will do soon). At the beginning of `getEpisode` , the state needs to be reset to the initial state where no edge is selected

Reply:
Should be solved by Moritz' branch.